modules/dhcp/dhcp_api.rb

```diff
@@ -62,15 +62,51 @@ def load_subnet_data
     end
   end
 
+  # Deprecated, returns a single record
   get "/:network/:record" do
     begin
       content_type :json
 
+      logger.warn('GET dhcp/:network/:record endpoint has been deprecated and will be removed in future versions. '\
+                  'Please use GET dhcp/:network/mac/:mac_address or GET dhcp/:network/ip/:ip_address instead.')
+
       load_subnet
       load_subnet_data
 
       record = server.find_record(@subnet.network, params[:record])
-      log_halt 404, "DHCP record #{params[:network]}/#{params[:record]} not found" unless record
+      log_halt 404, "No DHCP record for #{params[:network]}/#{params[:record]} found" unless record
+      record.to_json
+    rescue => e
+      log_halt 400, e
+    end
+  end
+
+  # returns an array of records for an ip address
+  get "/:network/ip/:ip_address" do
+    begin
+      content_type :json
+
+      load_subnet
+      load_subnet_data
+
+      records = server.find_records_by_ip(@subnet.network, params[:ip_address])
```

Comment:
Yeah, we could issue an 422 if you queried the endpoint with something that doesn't look like an ip. I'd suggest doing this in a different PR, though.